### ISSUE TITLE
Next steps for the Open Code of Conduct

### ISSUE THREAD

**Issue by bkeepers** [todogroup/opencodeofconduct#84]
Thanks to everyone who expressed concerns about the language in the proposed //TODO Open Code of Conduct template. 

We got a lot of constructive feedback. We also received complaints related to “reverse-isms” and claims of racism and sexism. This template aims to equally protect against all forms of discrimination, and to protect people who are typically the target of harassment in open source.

## Next Steps

We are going to take a step back and pause activity on this project for now. We are consulting with stakeholders, community leaders, and legal professionals. We want to make sure any further changes are done carefully and thoughtfully. 

You are welcome to fork our code and leverage those aspects of the text you like. You are also welcome to watch the repository and get notified when we reactivate the project. In the meantime, we want to thank everyone for the help in getting us to this stage.

Our primary interest is in creating healthy and productive open source communities, and a code of conduct is an important tool for accomplishing that. Thanks for your concern and patience. Stay tuned.

**Comment by atarzwell** [comment]
> We are going to take a step back and pause activity on this project for now. We are consulting with stakeholders, community leaders, and legal professionals. We want to make sure any further changes are done carefully and thoughtfully.

Will this be in effect during this time? 

Will the official github blog post continue to promote it as a "recommended and ready to use" CoC during this time?

I appreciate that you are listening to the community, but my concern is that this is just an attempt to shut down the conversation without having to address the concerning portions.

**Comment by CiaranLaval** [comment]
Thank you. As someone who works in a sector whereby I simply cannot knowingly be associated with projects that discriminate against people on the grounds of race, gender and sexual orientation, I appreciate the pause and wish you all the best going forward.

I also have to offer an apology, I decided last week that I needed to take a closer look at GitHub for my own interests and then this storm erupted, which is typical of my luck lately!

**Comment by CoWinkKeyDinkInc** [comment]
When should we expect the rewrite to happen?

**Comment by Addvilz** [comment]
It is sad to see this, basically, derailed by "few bad apples", however, community outrage about this hasn't been completely without reason.

However, as a shameless act of self-promotion (I hope TODO will forgive me), I did start working on https://github.com/openpolicy project, that could be a alternative while also covering more "ground" and under open governance model - if anyone is willing to direct their contributions there while this is on hold, your are welcome to do so.

**Comment by giulioprisco** [comment]
Re "This template aims to equally protect against all forms of discrimination, and to protect people who are typically the target of harassment in open source."

A template that equally protects against all forms of discrimination will equally protect everyone, including "the people who are typically the target of harassment in open source."

I recommend not to make things artificially complicated. The spirit of the template should be that insulting or harassing people based on their race, gender etc. is never OK and always punished. That protects everyone, while adding the "reverse -isms" bit protects some people but doesn't protect others.

**Comment by mtman** [comment]
> That protects everyone, while adding the "reverse -isms" bit protects some people but doesn't protect others.

The statement _legitimates_ selected (yet not defined) genders and races to racist and sexist behavior. Therefore, the COC is essentially sexist and racist, _not_ only protective of some.

**Comment by davezawislak** [comment]
#MoreEqual

**Comment by k-zed** [comment]
Terrible news... looks like this would've been too civilised for the open-source types.

It's really unfortunate that github is back-pedalling from something that looked pretty good (compared to the culture it's coming from).

**Comment by giulioprisco** [comment]
re "back-pedalling from something that looked pretty good" - Pretty good, you say? _Explicitly condoning_ insults and harassment directed at second-class participants is "pretty good?" Really now.

**Comment by k-zed** [comment]
Cisgender white males are not second-class participants, and are not systematically oppressed in any way (not even by things like affirmative action). The "reverse-isms" bit would be useful to shut down unproductive complaining by those who don't like that their privileges aren't putting them before the others by quite as much as they used to.

**Comment by davezawislak** [comment]
Discrimination upon anyone for just being a member of a class is discrimination. No one is more equal.

**Comment by jardaroh** [comment]
It is sad to see people that have a wanton lust to discriminate against particular groups of people. Though I am happy to see that GitHub may yet reconsider their stance to support this type of behaviour.

**Comment by Naouak** [comment]
> Cisgender white males are not second-class participants, and are not systematically oppressed in any way (not even by things like affirmative action). The "reverse-isms" bit would be useful to shut down unproductive complaining by those who don't like that their privileges aren't putting them before the others by quite as much as they used to.

Do you know that your comment is racist and sexist. From the moment, you choose to use race, gender or sexual orientation/identity as a critter, you become discriminatory. Discrimination is not always bad but when the critter used for discrimination should not be used for choice, it becomes bad. 

Does any women, non-white person, non straight person has the right to harass me because I'm "cisgender white male"? Hell no! No one has the right to harass. This should be it. I don't care what is your sex, your religion, your ethnic, your sexual orientation. You are harassed, this is bad. 

You are harassed and have privileges ? I don't care, being harassed is still not fine whether you have privileges or not.

**Comment by mtman** [comment]
@k-zed It is your far-left theory that a white (asian as well?) male systematically oppresses anyone just by being successful. When we look at objective [hard data](http://edition.cnn.com/2015/04/13/opinions/williams-ceci-women-in-science/) regarding privilege, we can find out that a woman has 2-to-1 (or even 4-to-1) advantage when applying for a job in STEM fields just because of her gender.
Please stop being racist and sexist and act civilized.

**Comment by askl56** [comment]
What's wrong with everyone being equal? I thought that's the meaning of the word equality, and I don't understand why it's a controversial idea. No reverseisms, just don't be unpleasant to each other.

![0f2a72acfea2544a6971114204a76c93](https://cloud.githubusercontent.com/assets/3062663/9150979/5c0a4dce-3dea-11e5-93de-c79020f98cb6.jpg)

**Comment by k-zed** [comment]
There is nothing wrong with everyone being equal - that's what we're trying to fight for.

But it's not currently the case. There is no equality on tech (or anywhere else) - that's just the current state of things.

**Comment by CiaranLaval** [comment]
There are a couple of points at play here that seem to be converging when they really shouldn't.

Nobody should be being discriminated against based on their gender, race or sexual orientation. Unfortunately the reverseisms give the impression that it is ok to discriminate on these grounds, I'm sure that's not the intent but that is the impression.

There are definitely groups in society who are under represented in technology, that needs to be recognised and addressed and at times that will make for some uncomfortable conversations, which is fine as long as they don't become abusive.

**Comment by giulioprisco** [comment]
k-zed, if we are fighting for everyone being equal, the code of conduct should reflect that. The reverse -isms part doesn't. On the contrary, it explicitely condones discrimination.

**Comment by JohnAppleSeed-** [comment]
I would just like to remind everyone of the definition of bigotry:

**Bigotry, noun - intolerance toward those who hold different opinions from oneself.**

So just remember, **you have to respect my opinion no matter what, or you're a bigot**.

**Comment by shreeve** [comment]
This is a complete waste of time... political correctness gone amuck.

**Comment by CoWinkKeyDinkInc** [comment]
@JohnAppleSeed- You misread the definition.

**Comment by mtman** [comment]
@k-zed Well there is equality of opportunity and equality of outcome. We rightists simply don't mind if tech or marketing will be dominated by 80% females or males as we assume it is simply their field of interest. Not everyone in tech is a leftist, please try to respect that.

**Comment by keisisqrl** [comment]
Meanwhile over here in reality, the rest of us will be working to make sure that there are not additional factors outside of field of interest excluding marginalized people from tech. Because it is simple fact that it is not the only factor. 

The evident primary controversial clause is important. Complaints of oppression of majorities, of tone, even of criticism of oppressive behavior or expression or enforcement of boundaries are nearly *always* used solely to shut down marginalized people. 

The fact is that tech at large is suffused with privilege, and the community is not generally inclined to analyzing itself in that fashion. It is critically important to ensure that marginalized people are safe in this community, particularly in the open source community which prides itself on inclusion and the ability of anyone to contribute.

**Comment by JohnAppleSeed-** [comment]
@keisisqrl 

>The evident primary controversial clause is important. Complaints of oppression of majorities, of tone, even of criticism of oppressive behavior or expression or enforcement of boundaries are nearly always used solely to shut down marginalized people. 

This is clearly coded racism. Marginalized people can be white straight males too.

**Comment by keisisqrl** [comment]
Sure. A person may be privileged in some ways and marginalized in others. It's all relative.

**Comment by sstjohn** [comment]
I'd still like to know why Github et al don't believe gays and Jews deserve protection from reverse -isms.

**Comment by samis** [comment]
Aren't reverse -isms equal to -isms?

**Comment by mtman** [comment]
@keisisqrl It is nice that you are interested more in reality than theories. But it is me who linked any hard evidence here so far, and it proves a privileged white male as a stereotype.

**Comment by mooreniemi** [comment]
The lack of understanding of what reverse-isms are or are not is telling. Equality is nothing without fairness, and fairness would require recognizing that some systems of oppression exist (against women, against POC) but some prejudices are _not_ systematic ("I hate men", "fuck white people"). The former are the ones we need to remediate as a society, the latter are rude to receive as an individual but will not impact your ability to survive in society.

**Comment by JohnAppleSeed-** [comment]
@mooreniemi 

You're racist. You think it's okay to harass white people but not others. This is racism.

You need to stop pretending like it's not racism. You support racism, but don't like to call it racism.
You think racism is okay.

**Comment by Benguin** [comment]
My friends and I would not use Github under this CoC; the part about reverse-ism is a horrible step in the wrong direction, regardless of intent, and other parts of it are just /silly/ rules to thrust upon a community you wish to behave like adults: "don't pretend to hug each other without permission!!"

I can only assume that the part about reverseism is there because you think it protects minorities from someone who would try to harass a a member of such minority by claiming that the minority is harassing /them/. But, the fact of the matter is that /minorities are perfectly capable of harassment, too/, and harassment of someone who doesn't belong to a minority, by someone who does, does indeed happen.

 By including that bit that essentially says "we won't tolerate complains about harassment if you think you're being harassed because you're a cisgendered white male", you're actually /neglecting/ your responsibility to protect /all/ of your users equally, which you claim to be your intent.

**Comment by logannc** [comment]
@mooreniemi

>The lack of understanding of what reverse-isms are or are not is telling. Equality is nothing without fairness, and fairness would require recognizing that some systems of oppression exist (against women, against POC) but some prejudices are not systematic ("I hate men", "fuck white people"). The former are the ones we need to remediate as a society, the latter are rude to receive as an individual but will not impact your ability to survive in society.

Systemic prejudices gain their power from their pervasiveness and lack of negative feedback from peers. These newer, less systemic prejudices gain their power from their increasing weight among people in charge due to the perceived prejudice in rebutting them.

To be more explicit, the discriminatory power of a prejudice can be thought of as `Power = frequency*weight/backlash`. Being prejudiced against women happens frequently (to various degrees) with a relatively low weight given to such opinions and average backlash. Being prejudiced about cis-gendered or white people or, God forbid, a white cis male, has less frequency but has a *huuuge* weight given to it. And, as these rules of conduct would enforce, they would have virtually no backlash.

Think about the event at that Python conference a while back. People lost their jobs over a 'dongle' joke. The guys that made the joke *didn't even know she was there*. If a woman had made a similar joke in reverse, no one would do anything about it. But because they put greater weight on her opinion than theirs, they were severely punished. (In this case there was a backlash against the woman, but only due to widespread media coverage.)

So, I think that you're wrong. We cannot allow some prejudices to take precedence over others. And I think you're wrong because these non-systemic issues *will* impact your ability to survive in society due to the increasing and disproportional weight given to them. They just happen in different ways. Systemic issues occur as the sum of small harms while non-systemic issues generally materialize as infrequent but more serious individual damages.

------

That said, I find that this conversation is necessary at all is ridiculous. I'm barely an adult so I find my worldview of society colored by my not-too-distant view as a child, where adults hide things from children and tell us children to get along our we'll be punished. 

And it is thoroughly pathetic that with adults can't just fucking get along. I mean, come on people. *Grow up.* Don't discriminate (but don't be too sensitive as to consider everything discrimination), don't be mean, don't take jokes too seriously, learn to take criticism (but with a grain of salt), and don't attribute to malice what is likely just ignorance.

Just get along, dammit, or I'm telling all of your mothers.

**Comment by whackashoe** [comment]
I use Github for code, not social justice politics. If this continues I'll be migrating everything off of Github. I don't even see the need, if someone is harassing people just allow them to block that person. Problem solved without loads of bullshit. Do you want Github to be a place where you share code, or a place where perpetual victims can try to find fault with anyone who thinks different to them?

**Comment by zero808** [comment]
>Do you want Github to be a place where you share code, or a place where perpetual victims can try to find fault with anyone who thinks different to them?

@whackashoe you answered your own question.

**Comment by Anonyfox** [comment]
> I use Github for code, not social justice politics. If this continues I'll be migrating everything off of Github.

Very much this.

**Comment by mjblack** [comment]
@whackashoe +1

**Comment by rr-** [comment]
>I use Github for code, not social justice politics. If this continues I'll be migrating everything off of Github. I don't even see the need, if someone is harassing people just allow them to block that person. Problem solved without loads of bullshit. Do you want Github to be a place where you share code, or a place where perpetual victims can try to find fault with anyone they disagree with?

This. First StackOverflow with their celebrating rights to homosexual marriages, then GitHub with this. I didn't become a neckbeard to deal with such social matters...

**Comment by CoWinkKeyDinkInc** [comment]
It is good that you guys are trying to fix this and hopefully remove the whole reverse racism thing from GitHub and the OCoC that almost everyone doesn't like; what I don't understand is why GitHub would implement this when it has been known to be flawed in some important parts.  There was reverse racism, the high-level and/or confusing language, and unnecessary clauses and rules.  Ultimately GitHub does not need a code of conduct that mentions physical contact in the website rules and other really specific groups.  
All it really needs as a website where developers can contribute to open-source software is for people to keep a calm, casual atmosphere and to not unfairly discriminate people.  People who are doing that and being annoying will be blocked and reported anyway.  It brings up so many issues that don't exist here that undeliberatly shows what ideals that people should have here; ideals that have nothing to do with cooperation or programming.

**Comment by zero808** [comment]
> what I don't understand is why GitHub would implement this when it has been known to be flawed in some important parts

@CoWinkKeyDinkInc To cater to the permanently offended crowd.

**Comment by juanplopes** [comment]
Good to see we have **all** sides well represented discussing this issue. I see a lot of white dudes and fictional characters in the avatars here, which are obviously the two most oppressed groups in this industry. I stick with the white dudes. I'd identify myself with the fictional characters if I could, but sometimes I feel they're a very ethnocentric group, and being a white dude is not that hard after all. I just have to accept that my biological traits don't give me the ability to make accurate decisions under pressure. You know, hormones and stuff.

Anyway, people, keep up the good work defending ourselves against the threat of the SJW! We're here to write code, and that's itself a hard task. We don't have the time for such politically correct bullshit of making our white-and-fictional-friendly space also friendly to other groups. "white-and-fictional-friendly" is already a long enough word.

**Comment by TETYYS** [comment]
>I use Github for code, not social justice politics. If this continues I'll be migrating everything off of >Github. I don't even see the need, if someone is harassing people just allow them to block that >person. Problem solved without loads of bullshit. Do you want Github to be a place where you share >code, or a place where perpetual victims can try to find fault with anyone they disagree with?

**Comment by mach-kernel** [comment]
I am all for people being able to go on a site such as this to be able to read code in a manner that is not polarizing nor insulting. I am very upset, however, that GitHub is going censor crazy in an attempt to please certain kinds of internet users that visibly seek conflict. I am a software engineer, I come here for the software. I don't think that places for learning or exchanging thoughts on engineering is a place to try out your latest social experiment to ban words that [some] find offensive. 

I personally use `retard` as the opposite of `advance` commonly when talking about engine timing on ECU tunes that I write with my friends. If I had a commit that said `retard timing`, would that be grounds for removing my repo? Ridiculous.

PS: Gitlab won so hard on the innovation side.

**Comment by f4w5yhq6z7bo** [comment]
May I recommend the No Code of Conduct (NCoC):

https://github.com/domgetter/NCoC

Read more here:

https://www.reddit.com/r/KotakuInAction/comments/3fr5d2/one_way_to_fight_back_against_entryist_sjw_codes/

**Comment by eevee** [comment]
I'm at least glad that GitHub is _trying_.  Please continue to do so.

@mach-kernel This is disingenuous and you know it; the takedown in question was not using the word in that manner.  There are valid criticisms to be made about the situation, but semantic games are not among them.

**Comment by zQueal** [comment]
> If I had a commit that said retard timing, would that be grounds for removing my repo? Ridiculous.

This is really what it comes down to. Words almost always have a meaning outside of what's seen as politically incorrect.

> This is disingenuous and you know it; the takedown in question 

There's absolutely nothing disingenuous about it as he's not even talking about the takedown in question... Re-read his comment.

Fact of the matter is, is that regardless of how it's used it's going to soon be hitting a filter and it can cause huge cockups for a great number of people. Things like this have absolutely no business in development. Github is meant to streamline VCS and help developers contribute positively to open source, and closed source projects. Not to become a bastion of politically correct pull requests. 

_Rediculous._

**Comment by vexorian** [comment]
Hey, somebody actually linked to KotakuInAction. That should be a good example or where the opposition to the reverse isms clause is coming from.

We have commenters here claiming that providing a safe environment for marginalized peopple has "no business in development".  I think this highlights how important it is to adapt a code of conduct. How easy it is for a site's culture to drift towards discrimination if not put in check.

Commenters in this show exactly why the reverse ism clause is necessary. They oppose even the existence of a code of conduct so if a CoC is put in place the first call to action would be to flood reports for trivial things like the reverse isms. Just like how we now have fake accounts brigading this comments thread to post links to NCoC. A CoC without the reverse isms clause would be at best ineffective in matching the stated intentions of the TODO group and at worst do the complete opposite.

**Comment by f4w5yhq6z7bo** [comment]
@vexorian 

> We have commenters here claiming that providing a safe environment for marginalized peopple has "no business in development". I think this highlights how important it is to adapt a code of conduct. How easy it is for a site's culture to drift towards discrimination if not put in check.

This is an instance of Kafkatrapping, a logical fallacy.

http://esr.ibiblio.org/?p=2122

You lose by default.

**Comment by jayyoung** [comment]
I think some kind of set of values, rules or guiding principles is great. It's good that GitHub is paying attention to these things. But it should be kept clear, fair and concise. The proposed CoC is so steeped in purposely ambiguous, wink-wink-nudge-nudge language and weasel wording that it's difficult to understand what it's trying to get at, and as a result I can just see unlimited potential for boring endless pages of rules lawyering over it right now. I don't understand why anything more complex than a few lines, ultimately summing to "Be excellent to each other" is required. Why start with something so complex.

**Comment by Benguin** [comment]
Just to clarify; I'm not against a CoC at all; I'm against /stupid/ CoC's which treat users like children, and which treat users inequally as this one does.

**Comment by JohnAppleSeed-** [comment]
>We have commenters here claiming that providing a safe environment for marginalized peopple has "no business in development".

This isn't what's happening and you know it. We have people actually defending racism under the guise of social justice. You can try to justify or characterize it however you want but you're a racist and a bigot. Racism is when you discriminate against a race regardless of their percieved privilege.

 The fact is white privilege is just a white stereotype. In nazi germany Hitler won much support by characterizing the Jews as the privileged race. 

The social justice warriors need to stop pretending they aren't racist.

**Comment by eaton** [comment]
> This is an instance of Kafkatrapping, a logical fallacy.

Incorrect. It's pointing out that the opposition to the CoC is not coming from concerned individuals who worry about implementation details, but rather people who *disagree that providing a safe environment for diverse participation is a good goal at all*.

**Comment by f4w5yhq6z7bo** [comment]
@eaton 

Incorrect. It's literally saying that opposition to the CoC justifies the CoC. It's as textbook an example of Kafkatrapping as can be found anywhere.

**Comment by CiaranLaval** [comment]
@eaton that is simply incorrect. Hardly any of the comments here have objected to providing safe spaces for diverse participation, most of the complaints are about discrimination on grounds of gender, race or sexual orientation.

**Comment by CiaranLaval** [comment]
>Let's also be real about people who threaten to leave github if the CoC is implemented. Have you seen the list of organizations that make the TODO group? Other code repositories will soon be implementing something similar.

Let's get real here, the organisations in the TODO group embraced the code of conduct before the reverseisms were added. 

Repositories may well implement something similar but for any professional organisation, it's a lawsuit waiting to happen, you simply cannot discriminate against people based on gender, race and sexual orientation in the manner the code of conduct that was last drafted did, it's wrong headed, bigoted and divisive.

**Comment by eevee** [comment]
I have mixed feelings about explicitly ignoring "reverse isms" in the CoC, but this very thread is full of comments [invoking them](https://github.com/todogroup/opencodeofconduct/issues/84#issuecomment-129060294) just to make trouble.

The current approach seems needlessly antagonistic, but there definitely needs to be some protection against people who're adept at following the letter of the law while flagrantly violating the spirit.

**Comment by eaton** [comment]
> Incorrect. It's literally saying that opposition to the CoC justifies the CoC. It's as textbook an example of Kafkatrapping as can be found anywhere.

No, it is saying that the *nature* of the objections to the CoC demonstrate why it is a useful and necessary document. If you can't understand the difference between those two statements, you shouldn't be throwing around links to logical fallacies and unilaterally declaring rhetorical victory.

**Comment by f4w5yhq6z7bo** [comment]
@eaton 

"The objections to a premise justify the premise being objected to." is Kafkatrapping, whether the focus is on the content, nature, tone, vocabulary, language, or something else. You can't make a logical fallacy go away by flavoring it. The fallacy is what it is.

**Comment by zQueal** [comment]
> We have commenters here claiming that providing a safe environment for marginalized people has "no business in development".

Not even close to what I said--but I guess thank you for putting words in my mouth. I'm saying Github acting like my kindergarten teacher has no business in development. I'm here to do a job, not to worry about if what I'm committing is hurting someone's feelings. If what I said offended you, speak to me about it. That's what adults do.

> It's pointing out that the opposition to the CoC is not coming from concerned individuals who worry about implementation details, but rather people who disagree that providing a safe environment for diverse participation is a good goal at all.

I'm totally __for__ providing a safe environment for diverse participation. That's why I participate in open source development. However, I'm wholly _against_ Github, as a company, moving against the community at large in an attempt to play moderator for those who get their feelings hurt by people they've never met before. If you _honestly can't handle_ curt criticism from random strangers then don't go into development--because I've never been so disrespected in my life then when trying to contribute to open source projects.

I support a level non-threatening playing field for anyone who wants to contribute; but what you guys are talking about here is plain censorship. Telling me, an adult, that there are words that I can and cannot use (aside from __obvious__ ones, such as slurs, which I would never use anyways) It serves to do nothing but forego progress. Words are _fucking_ magic. We use them to express ourselves, and emphasise points! Words like `retard` are also magic, because they can be used to express things other than profanities. Prohibiting them is doing nothing but removing the 1% of cases when it was being used incorrectly while simultaneously removing the other 99% when it was being used genuinely.

If you're being shitty, I'm going to tell you that you're being shitty--regardless of any factor that _you_ deem to be the cause, whether that be racism, sexism, fanboy-ism, or any kind of -ism. That's the way it needs to be--not some faceless corporation stepping in to play moderator. It needs to be up to the __community of developers__ to spot jerks who are committing these acts of -isms (or whatever you want to call it), and have them removed from the community by moderator action __after the fact__. But you don't put the majority of 3+ million developers on the defensive because someone might be offended by a word that has many other uses than as a pejorative. That's not even a sane option.

__That's__ the skinny of it. At least from my perspective. You're free to disagree, and I'm sure you probably will, but I have better things to worry about than if my use of `retard` is triggering you negatively. 

With that, I've said my piece. Do with it what you will.

**Comment by LegoStormtroopr** [comment]
@eaton you are a white-passing cis-male. Check your privilege when discussing social issues and dint presume to soak for minorities

**Comment by CiaranLaval** [comment]
Can the people who are for and against Gamergate please go and play somewhere else unless you genuinely care about GitHub, open source development and diversity. We really do not need that sideshow derailing this thread.

**Comment by JohnAppleSeed-** [comment]
It's hilarious that the ones that support the current CoC are a majority of white people. Like they don't have enough control, they have to make the rules for github too.

**Comment by LegoStormtroopr** [comment]
@CiaranLaval aged. Github is for code not social cause *de jour*.

**Comment by mjblack** [comment]
@vexorian 

>We have commenters here claiming that providing a safe environment for marginalized peopple has "no business in development". I think this highlights how important it is to adapt a code of conduct. How easy it is for a site's culture to drift towards discrimination if not put in check.

So to create a safe space you're willing to disparage a group of people? The funny thing is that last sentence rings so true to the nature of the CoC and why people oppose it.

> Commenters in this show exactly why the reverse ism clause is necessary. They oppose even the existence of a code of conduct so if a CoC is put in place the first call to action would be to flood reports for trivial things like the reverse isms. Just like how we now have fake accounts brigading this comments thread to post links to NCoC. A CoC without the reverse isms clause would be at best ineffective in matching the stated intentions of the TODO group and at worst do the complete opposite.

Yes, commenters like you do show us why we need a CLEAR and CONCISE CoC that does not disparage any one particular group, regardless of your personal feelings on the matter. 

> Hey, somebody actually linked to KotakuInAction. That should be a good example or where the opposition to the reverse isms clause is coming from.

Hey you're a block bot admin who has put people on the list for little to no reason... I'm going to go with that you're the perfect example of why we want those clauses gone.

**Comment by mach-kernel** [comment]
The very point that I was trying to make before being labeled as 'disingenuous' was the fact that the policy on retard and other words with similar context is a dangerous one to have, especially when the OSS community focuses on open-ness and freedom. I personally think that anyone who uses the word retard to offend or otherwise belittle another person is, well, disgusting. However, I think that they should have the right to say it - especially applicable to cases where someone can just wake up tomorrow and be offended that the word cat makes them feel mildly upset. What does this accomplish? 

Words are words. People give them context. Putting a filter on something never solves the true problem, and this is clear.

**Comment by eaton** [comment]
> @LegoStormtroopr: eaton you are a white-passing cis-male. Check your privilege when discussing social issues and don't presume to speak for minorities

I make no claim to speak on behalf of minorities; I'm just an open source person who's worked in a number of large communities, and helped implement (and refine) other codes of conduct. I've watched months of arguments by folks in favor of and opposed to CoCs of various types and descriptions, and seen what happens after they're actually implemented.

If you believe that my race and gender identity have blinded me to something that's germane to this discussion, I'm happy to discuss it. "Check your privilege" is a starting point, not an incantation to make people go away, and gamely parodying social justice catch phrases doesn't constitute an argument for or against a particular policy document.


> @mach-kernel: The very point that I was trying to make before being labeled as 'disingenuous' was the fact that the policy on retard and other words with similar context is a dangerous one to have, especially when the OSS community focuses on open-ness and freedom.

Just to clarify, do you mean that *the specifics of the proposed policy* are dangerous, or that having any policy on acceptable communication while participating in a particular project is dangerous? I can understand the latter, and think it's worth discussing further, but the idea that open source principles are incompatible with standards of conduct strikes me as a bit of a stretch.

**Comment by fragglet** [comment]
@bkeepers - Nothing much to add, just wanted to say thanks for the great work you're doing.

**Comment by CiaranLaval** [comment]
>If you believe that my race and gender identity have blinded me to something that's germane to this discussion, I'm happy to discuss it. "Check your privilege" is a starting point, not an incantation to make people go away, and gamely parodying social justice catch phrases doesn't constitute an argument for or against a particular policy document.

"Check your privilege" is absolutely designed to make people go away and is also an excuse to not address the socio-economic issue of what privilege really is, which is money, and the more you have of it, the more colour and race blind it becomes.

**Comment by burninglegion65** [comment]
Honest question. Is this supposed to be a US only thing? If not are you planning on checking which country a person is from to classify them first.

This should be far simpler than it has become and focus on the basics. Harassment is bad. Don't discriminate under any circumstances. 

GitHub is used globally and should reflect that.

**Comment by mkovarik** [comment]
I think this is a bunch of hullabaloo over nothing.

Marginalized people, women and minorities, are disproportionally targets of harassment, derision and exclusion. So an anti-bigotry policy automatically protects the marginalized more so than the privileged. The question that should be asked is as follows: does not acknowledging reverseisms actually protect the marginalized? I don't believe so.

Many argue that reverseisms don't actually exist. I understand the argument, and it may be correct, but I don't think its relevant either way. Even if reverseisms don't exist, anyone can still be nasty and toxic to anyone else for any reason ("Your hat is stupid!"). And it does not matter what the reason is. Any decent CoC includes a "be respectful" clause that implicitly covers all nastiness.

**Comment by LegoStormtroopr** [comment]
The thing about reverse-isms is, as pointed out, they fail to account for global issues.

For example, Japan is acknowledged as xenophobic (at least as much as any other country). If a Japanese and American developer are using racist slurs to each other thats racist. It doesnt matter if they are in America (where the American developer has social capital) or in Japan (where the Japanese developer has more power).

To suggest otherwise means that if our developers were traveling from L.A. to Tokyo saying the same slurs the whole flight at some point they switch from being racist to reverse-racist.

**Comment by LegoStormtroopr** [comment]
@mkovarik Who is marginalised is context specific. DjangoGirls is a predominantly female space (not a critique just an observation). Does that mean that in the DjangoGirls community men are the minority? Thus men are protected and you can't be reverse-sexist to women in that community?

**Comment by f4w5yhq6z7bo** [comment]
GitHub employs some openly, shamelessly, proudly bigoted personnel, which explains the culture motivating this Code of Conduct.

View the evidence here:

https://www.reddit.com/r/KotakuInAction/comments/3ga9sa/no_wonder_github_has_been_appeasing_sjws_recently/

**Comment by eaton** [comment]
I think the key issue about "reversisms" in the context of this issue is that it's too ambiguous to be helpful in the context of a code of conduct document. For example:

1. *Some commenters* seem to think it's referring to the idea that "racism" and racially charged language can only flow in one direction. i.e., racism against a white person is impossible because in the US whites are the dominant racial culture.
2. *Other commenters* seem to think it's referring to the idea that "positive steps taken to help under-represented groups sometimes draws objections by members of a dominant group who believe they are being 'reverse-discriminated against' because they don't receive the special benefits a given minority group does."

One of the most important aspects of a successful CoC is clarity. It doesn't necessarily have to enumerate every Bad Word and Unacceptable Action, but it should be very clear in the language it uses, and it should never introduce catch-phrases or terms of art that could cause more confusion than a plain and simple explanation.

For example, I'd agree that there's not much good that can come from encoding the *first* interpretation in a CoC, especially in a document meant to be re-used across cultures where the dynamics between different groups may be different. However, I would support a clause stating that "The CoC presupposes that positive steps taken to encourage participation by under-represented groups is not discrimination against any *other* groups." That, as I understood it, was the intent of the "anti-reversism" comments.

**Comment by Tynach** [comment]
There is so much here about whether or not we should metaphorically 'lower' people who are privileged, in order to make things more equal to everyone. But I think this is looking at the problem from the wrong angle, and fighting the battle (and true equality is always a worthwhile battle) from the wrong side.

Why not, instead of trying to bring the privileged down, we raise the unprivileged up? Would it not be better if the privileged retain all their privileges, and those privileges be opened up and given to those who don't have them? For true equality, would it not be best if we give all privileges to everyone?

Raising some people up while putting others down will *always* result in resentment and exclusion. It breeds bitterness and contempt on both sides. What's more, it starts a vicious cycle of putting others down and raising others up, never to reach equality.

Group A is privileged, Group B is not. Group A loses their privilege, Group B gains it. Now Group A is where Group B used to be, and vice versa... And as a result, under Group B's own logic, Group A deserves to take those privileges. And so the cycle goes on and on.

So many people are arguing about whether 'privilege' exists, or what it really means, or how it interacts with the dynamics of who can rightfully degrade who. But I don't see how that can ever work. Regardless of whether 'privilege' in this sense is real, and regardless what it means, isn't it in *everyone's* interest for *everyone* to have it?

**Comment by Tynach** [comment]
> However, I would support a clause stating that "The CoC presupposes that positive steps taken to encourage participation by under-represented groups is not discrimination against any other groups." That, as I understood it, was the intent of the "anti-reversism" comments.

@eaton, I'm cautious about that language. I understand what you mean and where you're coming from with it, and I 100% agree with the message, but I think the language is on the dangerous side.

The reason it's dangerous, is that you then need some mechanism for defining what constitutes as 'positive steps to encourage participation by under-represented groups'. If someone decides that such a 'positive step' would be to allow such a thing as your first given definition (which we both agree to be wrong), where, for example, women can discriminate against men while men cannot discriminate against women, there is a problem.

Under the wording you use, that 'positive step' is already presupposed to not discriminate against the well-represented group; in this example, men in comparison to the under-represented group of women. Therefore, there isn't room for debate about whether or not it is indeed a truly positive step.

I understand that this is nitpicking the exact words you used, and I agree with your sentiment's intention. However, I'm writing this because I have known people who would jump on the exact wording and use it to discriminate against people, pointing to the words for justification.

And while yes, that would indeed encourage them to participate... I fear that it is not the sort of participation any group would consciously want to attract.

**Comment by mkovarik** [comment]
@LegoStormtroopr Marginalization is, of course, the culmination of *all* social disadvantages that a person has for being in a group. Do you really believe that the existence DjangoGirls is enough to outweigh the disproportionate harassment, sexual violence, economic inequality (inside and outside tech), and other injustices women face?

You claim that racism is relative, and that people who are privileged in one culture may be marginalized in another. You are correct. Kurdish people, who might be considered "white", are certainly marginalized.

But when people decry "reverse racism", that's usually not what is being referred to. They are not advocating for the Kurds. Instead a teenager posts something mean about white boys on tumblr, and reddit loses its shit over literally nothing. Empowerment programs like Outreachy become popular, and men decry how sexist it is. Code of conducts are released and enforced, and suddenly the thought police are after you. A woman *dare* criticize video games? Threaten her with violence over twitter. That is the narrative that reverse racism brings with it.

**Comment by JohnAppleSeed-** [comment]
>They are not advocating for the Kurds. Instead a teenager posts something mean about white boys on tumblr, and reddit loses its shit over literally nothing. Empowerment programs like Outreachy become popular, and men decry how sexist it is. Code of conducts are released and enforced, and suddenly the thought police are after you. A woman dare criticize video games? Threaten her with violence over twitter. That is the narrative that reverse racism brings with it.

Wow that's a lot of strawmen for one comment. Sounds like you're basically saying white people can't stick up for themselves without being racist which is in its self a racist, bigoted, and narrow-minded viewpoint. Sounds like you're racist against whites to me.

**Comment by eaton** [comment]
> I understand that this is nitpicking the exact words you used, and I agree with your sentiment's intention. However, I'm writing this because I have known people who would jump on the exact wording and use it to discriminate against people, pointing to the words for justification.

Nitpicking the wording is exactly what a thread about a template CoC should be doing. ;-)

I think if there's a clearer way to express the idea—that the CoC implicitly rejects the idea that positive steps to encourage participation by Group A are not discrimination against Group B—it's worth pursuing. But if the CoC makes it clear that attacks on others' sexuality, gender, race, etc is unacceptable, the ambiguous interpretation you're concerned about would already be addressed.

> If someone decides that such a 'positive step' would be to allow such a thing as your first given definition (which we both agree to be wrong), where, for example, women can discriminate against men while men cannot discriminate against women, there is a problem.

This sounds like it might be a separate concern related to the idea of collaboration spaces specifically intended for particular groups. i.e., a project where all collaborators are expected to be of a certain faith. It doesn't really make much sense in the context that the CoC seems to be written for: general-purpose projects and communities that want to make things better.

**Comment by mkovarik** [comment]
@JohnAppleSeed-  The argument I made is that people use reverse racism to attack social justice movements and initiatives. You believe that is a strawman argument? That's a lot coming from someone who said "The social justice warriors need to stop pretending they aren't racist." Literally every comment you've made here proved my point.

**Comment by JohnAppleSeed-** [comment]
@mkovarik You're grossly mischaracterizing what's happening:

>Instead a teenager posts something mean about white boys on tumblr, and reddit loses its shit over literally nothing. 

This has nothing to do with the racist part of the CoC, this is where you begin to build your strawman. 

>Empowerment programs like Outreachy become popular, and men decry how sexist it is. 

Again, this has nothing to do with the racism in the CoC, it's part of a strawman argument you're building so that you *seem* right, when in reality this has nothing to do with anything going on here. 

>Code of conducts are released and enforced, and suddenly the thought police are after you. 

This is only partly relevant. No one is saying anything about thought police, they have legitimate concerns and you're just trying to build a strawman so you can paint any one who disagrees with your agenda as a racist. 

>A woman dare criticize video games? Threaten her with violence over twitter. 

Again, has nothing to do with what's going on here, it's just your way of bullying people on the internet to get your way.

**Comment by mkovarik** [comment]
>  Sounds like you're racist against whites to me.

~ JohnAppleSeed

**Comment by jayyoung** [comment]
>But when people decry "reverse racism", that's usually not what is being referred to. They are not advocating for the Kurds. Instead a teenager posts something mean about white boys on tumblr, and reddit loses its shit over literally nothing. Empowerment programs like Outreachy become popular, and men decry how sexist it is. Code of conducts are released and enforced, and suddenly the thought police are after you. A woman dare criticize video games? Threaten her with violence over twitter. That is the narrative that reverse racism brings with it.

Jesus. This is just.. what? Take a walk bro, step away from the computer.

**Comment by JohnAppleSeed-** [comment]
@mkovarik 

So when whites use racism to control the narrative it's bad but when you do it it's okay? You have a double standard.

You're sitting here calling everyone racist and when someone else calls *you* the racist suddenly they're the ones trying to shut *you* down?

And you don't see the hypocrisy?

**Comment by mkovarik** [comment]
I have never called anyone here (or rarely anyone anywhere else) a racist. Find a single example. You called *me* a racist, remember?

**Comment by JohnAppleSeed-** [comment]
You're being pedantic. You didn't use the word "racist" but you certainly implied that the people with a different opinion than your own are racist and sexist when you were building your straw man.

>They are not advocating for the Kurds. Instead a teenager posts something mean about white boys on tumblr, and reddit loses its shit over literally nothing. Empowerment programs like Outreachy become popular, and men decry how sexist it is. Code of conducts are released and enforced, and suddenly the thought police are after you. A woman dare criticize video games? Threaten her with violence over twitter. That is the narrative that reverse racism brings with it.

**Comment by mkovarik** [comment]
Anyway, the point I am trying to make is that reverse racism does not usually refer to comments like "I hate you because you are white", which would certainly be disallowed under any common sense CoC.

I believe that the CoC meant reverse racism in exactly examples I provided (except for the twitter example, which I apologize, is too extreme), which is why they are not strawmen arguments. 

But people really *do* criticize programs like Outreachy for being sexist against men (which it is not). So if Outreachy hosted code in GitHub, and if people complain of "reverse sexism" then it would make sense for GitHub to ignore these claims on the grounds that they are ridiculous (which is what I interpreted the reverseism policy to mean).

**Comment by carrchr** [comment]
It's really disheartening to see this discussion devolve into some GamerGate argument. People that talk about "reverse racism" as an actual thing should get a clue and just watch this video - https://youtu.be/dw_mRaIHb-M 
I applaud TODO group for taking a stand for what's right.

**Comment by mkovarik** [comment]
@JohnAppleSeed-  

> You didn't use the word "racist" but you certainly implied that the people with a different opinion than your own are racist and sexist when you were building your straw man.

You make too much personal assumptions of me. I actually *do not* believe that people who complain of reverse racism are themselves racist. I've in fact held similar beliefs (believing reverse racism was an important problem), and I do not see myself any more racist then as I do now.

My claim is that people use reverse racism as a way to attack social justice initiatives. That is a far cry from claiming anyone is racist.

For example, people who use "reverse sexism" to criticize Outreachy. They may argue that it is sexist because sexism against women in tech doesn't exist (or is exaggerated) because tech is meritocratic, and so Outreachy is providing opportunities to women that are unavailable to men. Women have an unfair advantage over men. The argument is obviously not sexist. But it is silly (and very, very flawed). And I believe these sorts of complaints are exactly the sort that tech organizations should ignore.

**Comment by carrchr** [comment]
@jayyoung +1 👍🏼

**Comment by eaton** [comment]
> if Outreachy hosted code in GitHub, and if people complain of "reverse sexism" then it would make sense for GitHub to ignore these claims on the grounds that they are ridiculous

Again, having helped draft CoCs for other communities and worked to refine and tweak them over time, I think that'd be a problem. "We will ignore ridiculous comments" just isn't something helpful in the context of a policy document. If there's a particular kind of ridiculousness that keeps coming up, figuring out how to address the class of ridiculousness more explicitly is usually a lot more productive.

Basically, "don't use catchphrases for contentious issues when a clearly written sentence or two will do."

**Comment by mkovarik** [comment]
@eaton Well, my assumption is that the reverseism clause of the policy document was in reference to this sort of complaint (empowerment project X is reverseist against group Y!), and not towards things like "I hate you because you are Y " (which I presume some people are interpreting it as).

Am I wrong?

**Comment by eaton** [comment]
@mkovarik, I took it the way you did, but the potential for misinterpretation is why I think a more explicit version would make more sense for a policy document. Particularly in the Drupal Code of Conduct, we found that sections using vague language or catch-phrases ended up being big problems, both in initial arguments about the document *and* in later incidents where interpretation of the CoC was needed by people doing community management work.

**Comment by mkovarik** [comment]
@eaton Then I believe that the CoC should come with clear examples of okay and not-okay behaviour. That is the best way to avoid ambiguities. People who that aren't familiar with SJ issues (many people on this thread, apparently) believe this to mean that privileged people do not deserve protection from harassment/bigotry, which absolutely not true.

**Comment by LegoStormtroopr** [comment]
@mkovarik you can see that this Code of Conduct as it stands says:

> We will not act on complaints regarding ‘Reverse’ -isms, including ‘reverse racism,’ ‘reverse sexism,’ and ‘cisphobia’

Which means, there are some people who are *explicitly not protected*, which implicitly says *these groups can be attacked without retribution*. Can you understand why people from these groups may be unhappy with this? Given that nerds were also probably bullied at school, can you understand why they might feel threatened by this?

Under the rules of this repo I can say:

> All white people are terrible programmers, and all men are the cause for all problems in technology. By marginalising these groups and preventing them from working in tech everything will be better.

And I have done literally nothing wrong. Is that ok?

**Comment by carrchr** [comment]
Nope you've got it wrong, @LegoStormtroopr. It means that "we" (whoever that is - the guardians of the repo, I guess) will not worry about white/male/hetero folks whining about reverse-isms. Because those are just made up things that only exist in there own mind, anyways. If some person is being a jerk/abusing you specifically then they will certainly take action.

**Comment by LegoStormtroopr** [comment]
@carrchr Firstly define a "reverse-ism" and secondly, what action could someone take against what I wrote? It is with in the guidelines as its only attacking "privileged groups".

And if I'm wrong, a lot of other people have made the same assumptions. Why do you think that is? Can you understand why this is irritating people?

**Comment by mkovarik** [comment]
@LegoStormtroopr Read @eaton reply. The CoC was never meant to mean "all white people are terrible programmers" is okay. Such language is vile, and there is a 0% probability that any community will tolerate it. CoC is meant to prevent bullying. ***Online abuse is never okay***. 

Reverseism refers to people (typically privileged) trying to undermine empowerment initiatives by claiming they are racist. That is generally what is understood by most SJ people (which is why the CoC is ambiguous, not everyone is aware of social justice concepts, except for egregious misrepresentations). We're all aldults. No one is claiming online abuse is okay.

**Comment by LegoStormtroopr** [comment]
So if someone says "all white people are bad", then I can claim that as racism and it will be dealt with as such?

**Comment by mkovarik** [comment]
Maybe. If someone says "You, LegoStormtroopr, are bad because you are white", then the answer is almost definitely (as @carrchr explained, abuse is actionable).

**Comment by LegoStormtroopr** [comment]
So if someone says "You, LegoStormtroopr, are bad because you are white" thats racist. But if they say "You, LegoStormtroopr are white." then in the same breath say. "White people are bad", thats not racist?

You keep saying "maybe" and "probably". To me both of those statements are definitely 100% racist.

**Comment by eaton** [comment]
@mkovarik:
> The CoC was never meant to mean "all white people are terrible programmers" is okay.

Just to be clear—if we're being pedantic, who knows what the intent of it was? Perhaps it WAS meant to say that! I don't believe so, but that's why clarity of purpose and language is important in a document like this. ;-)


@LegoStormtroopr:
> > We will not act on complaints regarding ‘Reverse’ -isms, including ‘reverse racism,’ ‘reverse sexism,’ and ‘cisphobia’

> Which means, there are some people who are explicitly not protected, which implicitly says these groups can be attacked without retribution.

Taken in isolation, I agree that the CoC can be interpreted that way. That's why I've said at several points in this thread that I'm in favor of a more explicit passage that spells out the kinds of claims that the CoC implicitly rejects, rather than relying on catch-phrases that are sometimes used to summarize those claims.

In the context of recent US political and social conflicts, those phrases ("reverse racism," "reverse sexism," and "cisphobia") have all been used to describe situations where previously disenfranchised groups, or those who've been the targets of discrimination, gained legal protection from discrimination or social mores changed to make discriminating them less acceptable. For example, growing social acceptance of same-sex marriage has been referred to by some of its opponents in this country as "heterophobia." Laws that ban employment discrimination against racial minorities have been described as "reverse racism" by their opponents, and so on.

It's that context of bad-faith rhetorical reversals that I believe the CoC passage relied on to preserve its meaning, and it's that context that many of its defenders assumed would be understood. That's a problem for any document in use by a broad community whose members might lack that context, and it's especially bad for a document meant to be a template. That's why I would much prefer an explicit passage that positively states what the CoC presupposes: that positive steps to assist or increase participation by group A are not "reverse discrimination" against group B.

**Comment by LegoStormtroopr** [comment]
> **In the context of recent US political and social conflicts...**

And there in lies the rub... there are lots of *us* who aren't from the US, and when we get dragged into US political affairs, we get kind of annoyed.

**Comment by eaton** [comment]
Precisely. That's why  I'm saying that the (contextually obvious, but ambiguous in isolation) meaning of the passage should be spelled out, rather than relying on catch-phrases that are very language and region dependent.

The concept that it's getting at is still important, even if the language that was proposed for it caused a lot of confusion.

**Comment by LegoStormtroopr** [comment]
Personally, I think an appropriate code of conduct for a programming community can be summed up as simply as possible as: "Write good code. Write nice words about other people code. Everything else is off-topic."

**Comment by mkovarik** [comment]
>  Write nice words about other people code

I don't think that's good policy. [Code is meant to be criticized](http://blog.codinghorror.com/content/images/uploads/2009/02/6a0120a85dcdae970b012877707a45970c-pi.png).Be mean to code. Be nice to people.

**Comment by eaton** [comment]
> Personally, I think an appropriate code of conduct for a programming community can be summed up as simply as possible as: "Write good code. Write nice words about other people code. Everything else is off-topic."

Unfortunately, as we've seen in many other communities large and small, that is often insufficient. It's like saying that "Comment well and don't commit bugs" is a sufficient coding standards document. For better or worse, different peoples' understanding of "nice" and "good" and "everything else" tend to differ wildly, and it's necessary to clarify, explain, and make sure that the intentions are as unambiguous as possible.

Just as some people spend a lot of time ironing out the details of a coding standards doc, some are motivated to iron out the details of a CoC. I'm sympathetic to "I think this passage has problems," or "I believe what the CoC says about that issue is going to make things worse" or whatever. But I'm pretty frustrated by the often-repeated refrain of, "CoCs are just a waste of time."

Software developers have vigorously, enthusiastically, proudly wasted time on things that not everyone cared about since before we were all born. No need to stop now. ;-)

**Comment by LegoStormtroopr** [comment]
@mkovarik Criticism can be "nice". There is a difference between, "This code is full of leaks and is shit, we aren't accepting this PR" and "This code introduces memory leaks in module XYZ, and needs be improved before we can pull it". Both hint at the same thing, the latter is "nicer". Someone could easily get insulted by the latter, but thats on them.

**Comment by Naouak** [comment]
I just read a lot of message and I see a big issue on the thread. Words are not defined and some defimitions seems to have a lot of variations.

For example, "racism". Racism, a word coming from french means "discriminating using race". That's it. There is no addendum to the definition for privilege or anything like that.

Now, can someone define "reverse-racism" please so we understand what it really mean? I know that the CoC tells you that you won't explain any social justice concept, but I beg for forgiveness as people here don't understand the same thing behind the same words.

**Comment by f4w5yhq6z7bo** [comment]
@LegoStormtroopr

I have pointed out the No Code of Conduct (NCoC) before. I am repeating it for your benefit.

https://github.com/domgetter/NCoC

Read more here:

https://www.reddit.com/r/KotakuInAction/comments/3fr5d2/one_way_to_fight_back_against_entryist_sjw_codes/

This is what a programming community needs, and absolutely nothing more. Anything else will be weaponized by social justice bullies. You can currently see this happening in this very thread.

**Comment by rr-** [comment]
Why do we even need a Code of Conduct? Why should we want a thing such as "No Code of Conduct" which sounds even sillier? If someone is a jerk, let's just ignore them and move on to better things... right?

**Comment by giulioprisco** [comment]
The No Code of Conduct (NCoC) is what I try to use in the communities that I manage or co-manage. Perhaps some parts of the text proposed by @LegoStormtroopr are unnecessarily confrontational, I would suggest to remove the justifications and just stick to simple, crystal clear rules.

An online community, like a society, needs only a few rules expressed in simple language, crystal clear, and strictly enforced.

"Insulting or harassing others is not permitted and will cause immediate expulsion" is a good rule, simple and crystal clear. It's also fair, because it treats everyone the same.

**Comment by f4w5yhq6z7bo** [comment]
@rr-

It's a necessary evil. I'll explain.

The very idea of a Code of Conduct at open source corporations and organizations was pushed by social justice bullies (some of whom can also be seen in action in this thread) as a sort of Trojan Horse to inject their racist, sexist, bigoted ideologies into these spaces. It's essentially sociocultural malware.

However, now that the idea of a Code of Conduct is part of the discussion, they won't let go. They'll keep pushing, and they'll try to frame any effort to not establish one as not caring about the community, or worse, as a desire to harm the community. We, in the community, may know better, but that's how they'll portray it to the public at large, especially through their corrupt press contacts.

The most practical method to neutralize this attack, in my humble opinion, is to establish a Code of Conduct, but one so similar to not establishing one as to be effectively equivalent to it. This prevents them from employing the psychological and rhetorical trick described in the preceding paragraph. They won't be able to publicly declare that concerned corporations and organizations refused to establish a Code of Conduct. They will have established one, just not one that the social justice bullies can abuse in any way. You may liken it to a null function to satisfy calls to a deprecated version of an API.

@giulioprisco 

Please differentiate the NCoC FAQ (which does use some colorful language and humor) from the NCoC itself (which is concise and civil). Please see:

https://github.com/domgetter/NCoC/blob/master/CODE_OF_CONDUCT.md

The FAQ is merely introductory documentation, and does not have to be included. The NCoC does the job.

**Comment by jayyoung** [comment]
>"Insulting or harassing others is not permitted and will cause immediate expulsion" is a good rule, simple and crystal clear. It's also fair, because it treats everyone the same.

Exactly. This is all you need, isn't it? What's the justification for having all these not-so-subtly implied instances of  "...oh and.." and "..but only if..." after that? This can surely only be a bad thing, because the above statement is *the sum total of everything a CoC should embody as it covers pretty much 100% of possible bad behaviour*, aside from maybe needing a clause about not causing technical disruption, but even then that's a form of harassment.

**Comment by CiaranLaval** [comment]
@eaton 

>Taken in isolation, I agree proposed passage in the CoC can be interpreted as a statement that claims of harassment will be ignored if they come from groups normally considered dominant. That's why I've said at several points in this thread that I'm in favor of a more explicit passage that spells out the kinds of claims that the CoC rejects, rather than relying on catch-phrases that are sometimes used to describe those claims.

I don't see how this will help. Once you get into a situation whereby you say you won't act on reports, you're batting on a sticky wicket.

Community owners and moderators already make judgement calls behind the scenes on whether or not an abuse report is valid. A statement that they won't act will discourage people from reporting abuse, which is not helpful to a community.

The reverseisms section is pretty much redundant, the rest of the code of conduct already makes clear that the community is an open and welcoming one to all, regardless of gender, race or sexual orientation, there's no need to add an unwelcoming clause such as the reverseisms, which undermine the overall message of the code of conduct.

**Comment by mjblack** [comment]
@CiaranLaval 

>The reverseisms section is pretty much redundant, the rest of the code of conduct already makes clear that the community is an open and welcoming one to all, regardless of gender, race or sexual orientation, there's no need to add an unwelcoming clause such as the reverseisms, which undermine the overall message of the code of conduct.

It is redundant except with the clause is specifically providing an exception to allow it. A lot of people have the misconception of what a reverse *-ism scenario is and why that entire section needs to be removed. The idea of a reverse *-ism is to give a minority preferential treatment over a majority solely because they're considered a minority. Not factoring for merit and experiences at all. The clause makes it so if you bring these concerns about this type of behavior happening in the project and they will be ignored.

**Comment by JohnAppleSeed-** [comment]
So basically it's okay to treat privileged people like shit?

This is bigotry plain and simple. People need to stop calling it "social justice" and just call it racism.

**Comment by f4w5yhq6z7bo** [comment]
Whenever a social justice bully makes a statement about white men, replace every mention of a white man with a Jew, repeat it back to him, and ask him whom it reminds him of.

That usually does the trick, if not for the social justice bully, then at least for the audience.

**Comment by hugofragata** [comment]
@rr- +1

**Comment by CoWinkKeyDinkInc** [comment]
Let's get back to the core reason why the reverse racism thing should be removed.  For a website and a professional business such as GitHub to contain text that discriminates certain people who want assistance from racial harassment is questionable and probably illegal.  If a lawsuit comes out of this it would probably be a more bigger controversy then the staff harassment thing that happened to this site a year ago.  This GG crap, who cares.  It has nothing to do with this issue.

**Comment by Sapein** [comment]
+1

Anywho I just want to state that I come to github for one reason, code, I don't come to have social issues and similar shoved down my throat. While I doubt I will have much effect on github, I am planning to move off ot it leaving a note in all of my respositories that I have moved it. If I wanted to deal with social issues I would go to where that is relevant, github is not that place. While the intention behind a Code of Conduct is good, the actual implimentation was bad and should be fixed. Whether or not Code of Conducts should be used or not is another debate entirely, in my opinion.

**Comment by KuraFire** [comment]
@bkeepers Is anyone from GH going to come in here and moderate this thread? Because @f4w5yhq6z7bo is clearly a GamerGater, constantly linking to KotakuInAction and being disingenuous and obnoxious and not actually trying to participate in the discussion in any kind of healthy manner. He also created that account at the height of GG and has only made 3 commits and nothing since (publicly, perhaps, but it seems a sock puppet account in any case, worthy of being blocked here). 
Back to need for moderation: there's various misconduct and misbehavior from their friends as well, which no moderator is calling out and so it's just an ongoing toxicity to this discussion.

I came here to make a statement of support for this CoC on behalf of the both the Modernizr and the One Web For All teams and organizations, representing thousands of less-privileged developers who benefit from Codes of Conduct being implemented, but right now this thread is a bit of a disaster zone with the teenage libertarian, disingenuous toxicity running amok.

(FWIW: the reverse -isms thing is a really good addition, but it can be refined and we'll have a suggestion for that in a different ticket, for consideration)

**Comment by f4w5yhq6z7bo** [comment]
@KuraFire 

I have participated in good faith and civility throughout this thread and I stand behind all of my statements 100%.

Feel free to challenge any claim I've made, and I will argue in its defense in more detail.

Thanks.

**Comment by giulioprisco** [comment]
KuraFire, this has been an interesting discussion so far, with different positions presented and defended. The first "toxic" post is yours. You don't even try to argue, but only attack others and call for bans.

**Comment by JohnAppleSeed-** [comment]
@KuraFire It's disingenuous of you to paint others you disagree with into a dark corner like you're attempting to do. You are the one who should be removed because you are blatantly attempting to shut down people who are disagreeing with your racist vitriol so that you can control the narrative of the conversation and push your racist authoritarian political agenda under the guise of "social justice". 

You are an openly racist bigot. Reverse racism is racism and reverse sexism is sexism and you are claiming to support these ideals. You are claiming to support discrimination against a whole group of people. There is no place for this vitriol in the discussion.

Not only are you psuhing an authoritarian racist agenda but you're openly bullying people which clearly violates the CoC.

>this thread is a bit of a disaster zone with the teenage libertarian, disingenuous toxicity running amok.

It's funny that you're against libertarianism but want to give private companies the go ahead to discriminate openly against non-minorities, which is a libertarian ideal in its self. 

Please don't attack other users here anymore.

**Comment by f4w5yhq6z7bo** [comment]
@giulioprisco 

You will find that this is a very common social justice tactic. Social justice is about as just as social security is secure.

**Comment by KuraFire** [comment]
@f4w5yhq6z7bo You don't have a good handle on what "in good faith" means, unless this is the best you can do:

From comment [#129059278](https://github.com/todogroup/opencodeofconduct/issues/84#issuecomment-129059278):

> You lose by default.

This is "in good faith"? No, this is disingenuous misconduct aiming solely to antagonize or incite anger into what was previously a reasonably healthy conversation.

> Let the record reflect that @vexorian is explicitly discriminating against minorities.

Here you completely misrepresent what @vexorian said by 1) taking it out of context and 2) assigning it new meaning that clearly wasn't there. No one but you and your fellow Gaters are buying this kind of behavior.

> Not if we have anything to do with it, they won't.

Here you are making a thinly-veiled threat of abuse and harassment for if you and your fellow Gaters don't get _your_ way, rather than a healthy and _just_ outcome by virtue of empathy and consideration for those who benefit from these CoC's being the victor.

> Besides, the same user goes on to explicitly discriminate against minorities right in the next paragraph, so that's a pretty good indicator of how much they value a safe environment or diverse participation.

Here you're lying again, misrepresenting someone completely.

I could go on, but there's nothing "in good faith" about your behavior here.

**Comment by Naouak** [comment]
@KuraFire stop making this aboit gamergate. The issue is whether the CoC is racist or not. We don't care about gamergate, we care about github's stance on racism.

**Comment by KuraFire** [comment]
@JohnAppleSeed- 

Wow, you're just a shining ray of acid.

You keep accusing people of being racist with _zero evidence whatsoever_, hell, you're accusing people of being racist when they aren't even talking about race.

You also misunderstand how racism, sexism, and discrimination work. And no, "reverse racism" is not racism, [as was already pointed out here](https://www.youtube.com/watch?v=dw_mRaIHb-M&feature=youtu.be). Reverse racism is not a thing (unless you actually have a time machine, in which case, cool).

**Comment by f4w5yhq6z7bo** [comment]
@KuraFire 

> This is "in good faith"?

Yes. Committing a logical fallacy loses the committer the debate by default. Doubling down on it (as was the case here) is just further evidence of the loss.

> 1) taking it out of context and 2) assigning it new meaning that clearly wasn't there

I did no such thing. He is discriminating against a minority. To find out details about which specific minority he is discriminating, people are free to read his full comment. I am obviously unable to quote the full comment when responding. It makes typesetting unfeasible.

> Here you are making a thinly-veiled threat of abuse and harassment

This is libel. Retract it immediately, or I will escalate this to GitHub management. I am not joking. I have had enough of your hate movement's tactics, and patience only goes so far. I will not stand for this anymore.

**Comment by KuraFire** [comment]
@Naouak I'm not the one doing that, I'm pointing out the gamergater who's trying to bring GG into this and saying this requires moderating.

**Comment by mjblack** [comment]
@KuraFire 

No you're being the toxic part of this discussion where your first post is coming in and attacking someone.

**Comment by JohnAppleSeed-** [comment]
@KuraFire 

> And no, "reverse racism" is not racism, as was already pointed out here. 

This is not a scientific source. Reverse racism is racism and you support racism. Plain and simple. You can characterize it however you want but you are a racist bigot.

You are toxic and a bully and only here to push a political agenda. You don't seem to care about "equality" at all.

**Comment by mjblack** [comment]
@KuraFire 

> Reverse racism is not a thing (unless you actually have a time machine, in which case, cool).

Using a comedian's skit to prove it doesnt exist, when the comedian is wrong on what it means, doesnt help your point. 

As for the toxicity of this discussion, up until you chimed in it was still civil.

**Comment by KuraFire** [comment]
I’m not “attacking” anyone, I'm calling out [the person who attacked Github with a bunch of libelous accusations and uses KiA to justify that](https://github.com/todogroup/opencodeofconduct/issues/84#issuecomment-129091418).

**Comment by mjblack** [comment]
@KuraFire 

> I’m not “attacking” anyone, I'm calling out the person who attacked Github with a bunch of libelous accusations and uses KiA to justify that.

Uhh see below...thats an attack. Instead of addressing the valid points you tried to attack his character.

> @f4w5yhq6z7bo is clearly a GamerGater, constantly linking to KotakuInAction and being disingenuous and obnoxious and not actually trying to participate in the discussion in any kind of healthy manner. He also created that account at the height of GG and has only made 3 commits and nothing since (publicly, perhaps, but it seems a sock puppet account in any case, worthy of being blocked here). 
Back to need for moderation: there's various misconduct and misbehavior from their friends as well, which no moderator is calling out and so it's just an ongoing toxicity to this discussion.

**Comment by KuraFire** [comment]
@mjblack he never made valid points, he spouted KiA links, threats, and libel.

**Comment by eaton** [comment]
> stop making this aboit gamergate. The issue is whether the CoC is racist or not. We don't care about gamergate, we care about github's stance on racism.

It's relevant because at least one of the active people in this issue is crosslinking it to GamerGate's subreddit, pointing to that subreddit to clarify their own positions. They are not discussing ways to improve the CoC document. Rather, they are advocating an explicit "anti-CoC" that protects serial harassers from repercussions. In software terms, they've jumped into node's issue queue to argue that people should use python.

In fact, at numerous points in this very issue they have gone off on tangents about the evils of Social Justice, linked to GamerGate articles in order to justify personal attacks on GitHub employees, and unilaterally "declared victory" for themselves in *an issue queue*.

**Comment by KuraFire** [comment]
@mjblack And you may disagree with or dislike Rahman's skit, that doesn't make it wrong. His point is based on the academic studies and generally-held academic position that racism = prejudice + power, and the examples Rahman mentions in his skit are exactly the things academia have found justify that position. Their stuff is just much less funny.

**Comment by f4w5yhq6z7bo** [comment]
@KuraFire 

You have been reported to GitHub. The issue may also be escalated above GitHub.

Libel is serious business, as you will find out to your detriment.

**Comment by JohnAppleSeed-** [comment]
I bet it's fun for the PC police to call all their critics racist and sexist and sit upon a moral high-horse of self-righteousness while pushing their authoritarian ideology under the guise of "social justice", in reality they're just as bigoted as the people they claim to be against. They people go around bullying people and making fun of people and think somehow they're the righteous ones.

GitHub should not pander to these racists. In the end equality will prevail.

**Comment by rr-** [comment]
C-can't we just hug it out? --oh wait

**Comment by mjblack** [comment]
@KuraFire 

> I’m not “attacking” anyone, I'm calling out the person who attacked Github with a bunch of libelous accusations and uses KiA to justify that.

The only problem is a lot of the language used in the tweets is reverse *-isms. Like... https://archive.is/6r6bR which is in the KiA article.

So if the CoC is creating a system of reverse racism by allowing it without penalty, which in itself is racism, would that not be creating a system of oppression?

**Comment by eaton** [comment]
> So if the CoC is creating a system of reverse racism by allowing it without penalty, which in itself is racism, would that not be creating a system of oppression?

Just out of curiosity, are you asking a good faith question about someone's views on how systems of oppression and disenfranchisement work? Because this is an important and worthwhile discussion, and even inside of circles that good ol' f4 would call "Social Justice," there are varying views on the matter.

Talking it out is not useless, especially if the goal is actually to arrive ad a shared understanding of each other's views and improve the CoC.

**Comment by carrchr** [comment]
+1 @KuraFire. I agree this thread probably should just get locked since it's becoming a left v. right shouting match, i.e. people that recognize the absurdity of "reverse racism" and why it is acceptable to ignore complaints about such versus people that somehow feel ignoring such complaints amounts to actual discrimination.

**Comment by f4w5yhq6z7bo** [comment]
May I also point out that any definition of racism that discriminates against certain races is itself racist, meta-racist, if you will.

Using a meta-racist definition of racism does not acquit you from charges of racism, if anything, it clinches the matter.

**Comment by mjblack** [comment]
@KuraFire 

> And you may disagree with or dislike Rahman's skit, that doesn't make it wrong. His point is based on the academic studies and generally-held academic position that racism = prejudice + power, and the examples Rahman mentions in his skit are exactly the things academia have found justify that position. Their stuff is just much less funny.

No, racism is not prejudice + power. That is what has been spoon fed to make reverse racism justifiable. You think it is ok to dehumanize a group of people because you perceive that they have privilege?

**Comment by KuraFire** [comment]
@mjblack:

> So if the CoC is creating a system of reverse racism by allowing it without penalty, which in itself is racism, would that not be creating a system of oppression?

Short answer: no, it would not.

Long answer: I get your concern here in the general sense, and it's a small nuance problem with the phrasing in the current CoC about reverse-isms. That does not mean, however, that the section on reverse-isms themselves should be taken out, because it is actually a really valuable addition to CoC's in general.

First, you'll have to accept the academic view on this issue, which is that reverse racism could only exist under conditions like the one Rahman jokingly paints with the time machine.  A more close-to-home, less-sci-fi version of that however would be Affirmative Action, or Positive Action, i.e. a system stipulating that a certain effort needs to be made to hire/attract people outside of the dominant group, to foster diversity.  These things are not "discrimination against (the dominant group, in this and most cases that would be cis straight white men)", they are rebalancing methods that break down some of the barriers that are currently held against all of the groups that are NOT the dominant one. I.e. they are systems to break down walls, not systems to discriminate.

**Comment by KuraFire** [comment]
@mjblack:

> No, racism is not prejudice + power. That is what has been spoon fed to make reverse racism justifiable. You think it is ok to dehumanize a group of people because you perceive that they have privilege?

Yes, yes it is, and no that's not about making "reverse racism" justifiable, and neither is it dehumanizing anyone.

Saying "Hey you, white man, stop punching down" _**does not dehumanize you**_.

**Comment by mjblack** [comment]
@KuraFire 

> Short answer: no, it would not.

Short answer: yes it would.

You're creating a system where if a white male complains about treatment by others in the project it will go largely ignored because they are white and male.

> Saying "Hey you, white man, stop punching down" does not dehumanize you.

So you assume all white men punch down? What a very bigoted mindset you have.

**Comment by Naouak** [comment]
@eaton if I follow your argumentation, if I post that : http://google.com , then the thread is also aboit google now.

Gamergate is off topic and should not be taken into account to decide whether this CoC is racist or not.  You even gave a good comparison why it's not relevant to talk about them.

Github is not supposed to be gamergate vs antigamergate battleground. This is not relevant to github and should not be discussed here.

@KuraFire The definition of racism according to dictionary is "discrimination based upon race". Don't use a made-up definition used by some people to justify their hate against white males.

**Comment by eaton** [comment]
> @Naouak The definition of racism according to dictionary is "discrimination based upon race". Don't use a made-up definition used by some people to justify their hate against white males.

1. a belief or doctrine that inherent differences among the various human racial groups determine cultural or individual achievement, usually involving the idea that one's own race is superior and has the right to dominate others or that a particular racial group is inferior to the others.
2. **a policy, system of government, etc., based upon or fostering such a doctrine; discrimination.**
3. hatred or intolerance of another race or other races.

The "prejudice + power" comment so many people seem to flip out at refers to the literal dictionary definition of racism.

**Comment by NuckChorris** [comment]
This whole stupid argument is why I made #69. I wish they hadn't thrown the baby out with the bathwater and closed a lot of otherwise quite awesome pull requests just because of shit like this.

**Comment by eaton** [comment]
> This whole stupid argument is why I made #69. I wish they hadn't thrown the baby out with the bathwater and closed a lot of otherwise quite awesome pull requests just because of shit like this.

I think you're correct. When the Drupal community went through the long process of drafting, discussing, and revising its CoC, it took a lot of distinct sub-discussions to iron out a lot of specific complex issues.

**Comment by mjblack** [comment]
@eaton I'm going to go with the U.N. definition of racism here, not the socjus one.

http://www.hri.org/docs/ICERD66.html

**Comment by JohnAppleSeed-** [comment]
>The "prejudice + power" comment so many people seem to flip out at refers to the literal dictionary definition of racism.

No it doesn't. And it implies that blacks can't have power, which is racist.

**Comment by Naouak** [comment]
@eaton you are misreading the definition. 

The definition talks about belief of superiority and building a system around it. Nowhere it says you have to have "power + prejudice" to be racist, it only says you can be racist by believing it. 

Let's note the third point of your definition, the one I pushed since the beginning of this thread : "hatred or intolerance of another race or other races".

**Comment by Christopher-Churnick** [comment]
@JohnAppleSeed- 

>No it doesn't. And it implies that blacks can't have power, which is racist. 

So wait, dictionaries are implicit now? Man, I'm old.

**Comment by KuraFire** [comment]
@JohnAppleSeed- 

> I bet it's fun for the PC police to call all their critics racist and sexist 

So you're the PC police now? Because in this thread, you're the one doing that _repeatedly_, against people who are… not doing that at all.

**Comment by JohnAppleSeed-** [comment]
@Christopher-Churnick  You're misinterpreting. 

Power+privilege is a phrase racists use. The phrase implies that non-whites can't be racist because they lack power and privilege. 

This is in its self racist because it implies that you believe non-white are incapable of holding power or being privileged. It's racist, bigoted, and narrow minded. 

No respectable dictionary defines racism this way.

**Comment by KuraFire** [comment]
@mjblack:  way to go, using a 55-year old understanding of racism. Nothing has happened in the past 55 years, nope, not a thing, no academic research or scientific publications on racism took place in the past 55 years at all, no sirree.

**Comment by JohnAppleSeed-** [comment]
@KuraFire 

The difference is that I'm actually pointing out real racists instead of engaging in personal attacks and bullying and trying to shut other people down by pretending that they're racist.

**Comment by KuraFire** [comment]
@JohnAppleSeed-: you're a little too full of yourself for me to continue this conversation. You aren't the arbiter over who is "the real racist" any more than I am, but at least I'm not using ridiculous retorts going "no YOU'RE the real racist" like we're in a kindergarten sandbox here. That's not adult behavior at all, it's behaving like a petulant child, a brat, and I'm tired of catering to your insistence on engaging at that level.

**Comment by TeddStriker** [comment]
@KuraFire 

What has changed in the last 55 years that allows us to label that definition as invalid? Sources please, not conjecture and appeals to feelings.

**Comment by f4w5yhq6z7bo** [comment]
We may simply have to sue these people in state and/or federal court. It looks like that's what it will take.

**Comment by Naouak** [comment]
@KuraFire I invite you to read @eaton definition:

> 1. a belief or doctrine that inherent differences among the various human racial groups determine cultural or individual achievement, usually involving the idea that one's own race is superior and has the right to dominate others or that a particular racial group is inferior to the others.
> 2. a policy, system of government, etc., based upon or fostering such a doctrine; discrimination.
> 3. hatred or intolerance of another race or other races.

You believe that white people have power and prejudice over any other race. If I read the definition, it correspond to the first part of the definition. You adhere to a racist doctrine.

**Comment by Christopher-Churnick** [comment]
@JohnAppleSeed- 

>This is in its self racist because it implies that you believe non-white are incapable of holding power or being privileged. It's racist, bigoted, and narrow minded. 

Just to play devils advocate here, I suggest we take a look at Nelson Mandela as an example of a person of colour achieving both power and privilege. From 1943 until 1964, Nelson Mandela was able to study law at University of Witwatersrand, become president of the ANC and tour the continent of Africa to drum up support against apartheid.  

He did all of this whilst under South African apartheid law. 

So, 2 questions:

1.) Was apartheid a racist system of oppression during the 20th century?
2.) Was Nelson Mandela powerful within that system?

**Comment by JohnAppleSeed-** [comment]
This thread is choc-full of **coded racism** and white men attempting to abuse their privilege and intellect by taking advantage of and shutting down those who aren't smart enough (because of lack of privilege) to defend them selves.

It's abusive to take and twist the definition of racism to somehow give the appearance of not actually being racist, when in reality their ideology revolves around the fact that they think some races are more equal than others. 

@Christopher-Churnick 

1.) Was apartheid a racist system of oppression during the 20th century?
Yes
2.) Was Nelson Mandela powerful within that system?
It would appear that way

**Comment by NuckChorris** [comment]
It's not that non-whites aren't capable of holding power or privilege. It's that they simply aren't holding it. They're fully capable, but because of historical reasons they haven't gotten that power or privilege.

So anyone who says accepting this reality is racist: shut the fuck up and actually think about it.

Yes, excluding reverse racism is honestly discriminatory and we shouldn't have that part. No, your opponents aren't racist.

Crap you're all acting like children here

**Comment by eaton** [comment]
One of the reasons it's useful to note that GamerGate members are participating in a discussion, and cross-promoting it on GamerGate discussion forums, is that every time they are involved _this is exactly how the thread plays out._

It's an often-repeated pattern of disruption, based on endless semantic arguments about the "real meaning of racism," personal attacks followed by monocle-popping outrage when they're called on it, and complete disinterest in the goals of the particular issue/discussion they've elbowed into.

It is really unfortunate that discussions they become involved in devolve into meta so consistently, but it's not the people who *notice their presence* that are causing the devolution.

**Comment by Aliababwah** [comment]
> I use Github for code, not social justice politics. If this continues I'll be migrating everything off of Github.

Another advocate of @whackashoe here.
